### ISSUE TITLE
 the feature of having the note window detached from the Obsidian main window

### ISSUE THREAD

**Issue by FeiLZX3Km** [nothingislost/obsidian-hover-editor#267]
May I ask if this plugin will be able to support the feature of having the note window detached from the Obsidian main window, floating independently on the computer desktop, and even supporting edge hiding? If so, that would be very convenient, as I would be able to take notes anytime without having to open the Obsidian main window.

**Comment by pjeby** [comment]
There are several ways to do this.  First, if the hover editor is focused, you can use the keyboard commands "Move current tab to new window" or "Open current tab in new window", which will move or duplicate the hover editor in a standalone window, respectively.

Or, if you prefer to use the mouse, show the hover editor's "view header" by clicking the boxed "v" in its title bar (unless you have turned on the "show view headers by default" option in hover editor settings, in which case you won't need to).  Either way, with the view header visible, you can drag the icon that appears on the left side of the header out to your desktop and drop it there or on another app, and the hover editor will pop out to a detached window.

(Note that these actions are available for *all* tabs, by the way, not just hover editors.  But with regular tabs you can drag them by the tab, even if you don't have their view headers visible.)

**Comment by tu2-atmanand** [comment]
+1 for this feature.

All the above ways mentioned by @pjeby are all valid and can be used to accomplish this task. But again this will give a little different experience than how Hover Editor is providing right now, also UI-wise. I actually wanted something like shown in below gif, where user can simply drag the opened Hover Editor/HoverPopover outside of the main window. This feature again can boost the functionalities of this plugin for specially those who have multiple monitors.

![Image](https://github.com/user-attachments/assets/c587a552-3dba-4d42-9f9d-e19feafee555)


I believe one challenge in having this feature inside plugin maybe the limited APIs provided by Obsidian or maybe the way Electron framework was used by Obsidian, it might not be possible at current stage. Or this may be very difficult to achieve in Electron itself. Would love to know, which case is it, which might restrict us from having this feat in this plugin.

**Comment by pjeby** [comment]
Yes, the way Obsidian does it (drag out tabs to new windows) is already the best it can get with Electron.  You can't tell the mouse is going outside its window unless you are doing a specialized drag and drop, and when you do drop, you can't really tell *where* it was dropped, which is why when you drag out a tab it ends up hovering near where you had it to start with.

**Comment by opperstar** [comment]
I have the same idea as you. I want to use the same UI outside the window. I found a video that achieved this function, but it is for the use of memos. I don't know how to program. I hope this can be helpful to you. If you manage to implement it, please teach me!!! The following are the 
video link: 
https://www.bilibili.com/video/BV1BL411P7Tx/?spm_id_from=333.337.top_right_bar_window_default_collection.content.click&vd_source=42262eeaa5229777950408f1bedfb55a

CSS link.:
https://github.com/PKM-er/Blue-topaz-example/blob/main/20-Diary/.obsidian/snippets/memos.css

**Comment by opperstar** [comment]
@tu2-atmanand 
I have the same idea as you. I want to use the same UI outside the window. I found a video that achieved this function, but it is for the use of memos. I don't know how to program. I hope this can be helpful to you. If you manage to implement it, please teach me!!! The following are the
video link:
https://www.bilibili.com/video/BV1BL411P7Tx/?spm_id_from=333.337.top_right_bar_window_default_collection.content.click&vd_source=42262eeaa5229777950408f1bedfb55a

CSS link.:
https://github.com/PKM-er/Blue-topaz-example/blob/main/20-Diary/.obsidian/snippets/memos.css

**Comment by pjeby** [comment]
You can use the "Move current tab to new window" command after clicking in a hover editor.  This will make it into a window that you can move anywhere on your desktop.  If you want it to always be above your other windows, use the command "toggle always on top".

In other words, Obsidian already has the functionality in the video above, you just have to use two commands, and you can assign shortcut keys to them if you want.

To be clear: here are the steps to convert a hover editor to a free-floating, always-on-top window, like the one in the video:

1. Click inside a hover editor to make it focused (unless it is already focused)
2. Use the obsidian command "Move current tab to new window" (this will pop it out into a free-floating window)
3. Use the obsidian command "Toggle always on top" (this will make it stay in front of other windows, even if the other window is focused)

You can of course also make a macro to perform these commands together if this is something you do often.  Just be sure to have enough of a delay between the two commands (say 500ms or so?) to allow the window to fully pop out, or it might make the wrong window be always on top.

**Comment by opperstar** [comment]
@pjeby thanks，I hope to keep hover-editor  UI interface unchanged. This is very important to me.😂😂

**Comment by tu2-atmanand** [comment]
Hii @opperstar!

Nice to know that you're also wishing the same functionality in this plugin. I havent gone through the resources you shared yet, neither the codebase of this plugin. (But ill be going through this repo, to get inspiration for few other things, I want in my plugin).

But specifically talking about the functionality we need here, from what my knowledge of the framework and Obsidian APIs, its kind of impossible thing to do at right now, or in other words few things are restricting us from achieving it in this plugin, as also nicely explained by @pjeby in above comments.

BUT..

As we all know, one of the most important functionality any note taking app should have is sticky notes. Like the one we have here using this plugin, where we are creating a small floating windows, which also remains always on top(obviously within Obsidian app itself) and we can add our thoughts to this floating window note. So its kind of working as the sticky note. But Obsidian always lacked this functionality.

However, I recently came across one new plugin, which is under review process right now and will be soon released in commnity plugins : https://github.com/Abdo-reda/obsidian-sticky-notes-plugin

It seems like @Abdo-reda has came up with some new solution, and has able to make this impossible thing possible. The plugin is exactly doing what is expected. It has few performance bugs and a lot of interesting features to come yet, like resizing and what not. But this new plugin has crazy potential. So, not exactly like HE UI, which we all love, but somewhat similar UI and most importantly the sticky note functionality will going to have through this new plugin.

**Comment by opperstar** [comment]
@tu2-atmanand you can try this way:
1. open new vault (It can be the folder in the previous path.) 
2. Download the theme pack you like
3. plugins: 
>hider (Hide the redundant interface, make it looks like a sticky note)
>global hotkeys (show/hide sticky note by hotkeys)
>electron window tweaker (window opacity percent)
[https://raw.githubusercontent.com/opperstar/MypPic/img/img/20250310085405318.png](url)

**Comment by opperstar** [comment]
@tu2-atmanand you can try this way:
    open new vault (It can be the folder in the previous path.)
    Download the theme pack you like
    plugins:
    hider (Hide the redundant interface, make it looks like a sticky note)
    global hotkeys (show/hide sticky note by hotkeys)
    electron window tweaker (window opacity percent)
  ![lie this](https://raw.githubusercontent.com/opperstar/MypPic/img/img/20250310085405318.png)

**Comment by tu2-atmanand** [comment]
> open new vault (It can be the folder in the previous path.)
> Download the theme pack you like
> plugins:
> hider (Hide the redundant interface, make it looks like a sticky note)
> global hotkeys (show/hide sticky note by hotkeys)
> electron window tweaker (window opacity percent)

Yeah, this is a nice workaround, but I dont think this will be required as the plugin which I mentioned, [Sticky notes](https://github.com/Abdo-reda/obsidian-sticky-notes-plugin), can able to give all this functionalities with a much better integrated experience(except the Electron tweaking one, as that will be too much to do in terms of customization and possible can lead to some bugs in your software). So, ill recommend instead of using this kinds of tweaks, its always better to use the verified and well integrated plugins directly. I am appreciating the Sticky Notes plugin is because its something Obsidian always should had, but first time I have seen someone had implemented it.
